### PR TITLE
[script.colorbox] 2.0.0

### PR DESCRIPTION
**Pull request by BADMS**
### Description

- Version 2.0.0 only for Krypton and above due to Pillow and maybe using new builtins not available in Jarvis or lower.
### Checklist:

- [X] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [X] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-scripts/blob/master/CONTRIBUTING.md) document
- [X] Each add-on submission should be a single commit with using the following style: [script.foo.bar] v1.0.0

Additional information :
- Submitting your add-on to this specific branch makes it available to any Kodi version equal or higher than the branch name with the applicable Kodi dependencies limits.
- [add-on development](http://kodi.wiki/view/Add-on_development) wiki page.
- Kodi [pydocs](http://kodi.wiki/view/PyDocs) provide information about the Python API
- [PEP8](https://www.python.org/dev/peps/pep-0008/) codingstyle which is considered best practise but not mandatory.
- This add-on repository has automated code guideline check which could help you improve your coding. You can find the results of these check at [Codacy](https://www.codacy.com/app/Kodi/repo-scripts/dashboard). You can create your own account as well to continuously monitor your python coding before submitting to repo.
- Development questions can be asked in the [add-on development](http://forum.kodi.tv/forumdisplay.php?fid=26) section on the Kodi forum.

### CODE REVIEW

**Comment by BADMS** [comment]
Is this the repo?

**Comment by BADMS** [comment]
Hello?

**Comment by enen92** [comment]
Colorbox seems to be on the Jarvis branch: https://github.com/xbmc/repo-scripts/blob/jarvis/script.colorbox/addon.xml

**Comment by BADMS** [comment]
Ok thanks, I *hopefully* have fixed commented issues, can you squash or this okay to add commit?

Regards,
dan

**Comment by BADMS** [comment]
Actually this has broken something ... please wait while I investigate :(

**Comment by BADMS** [comment]
Ok, all done. Better solution!
Hopefully you can squash all, or I'll just delete pull and resubmit, not up to speed on squashing via IJIdea..

File: script.colorbox/resources/lib/ImageOperations.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+    def filter(self, image):
+        return image.gaussian_blur(self.radius)
+
+
+class UnsharpMask(ImageFilter.Filter):
+    NAME = "UnsharpMask"
+
+    def __init__(self, radius=2, percent=150, threshold=3):
+        self.radius = 2
```
Comment (Tadly) [inline script.colorbox/resources/lib/ImageOperations.py]: I think you want:
`self.radius = radius`
here

File: script.colorbox/resources/lib/Utils.py [hunk 2]
```diff
@@ -0,0 +1,674 @@
...
+                return xbmc.translatePath(xbmc_vid_cache_file)
+            else:
+                filterimage = urllib.unquote(filterimage.replace("image://", "")).decode('utf8')
+                if filterimage.endswith("/"):
+                    filterimage = filterimage[:-1]
+                xbmcvfs.copy(filterimage, targetfile)
+                return targetfile
+                #return filterimage
+        except:
+            xbmc.sleep(200)
```
Comment (Tadly) [inline script.colorbox/resources/lib/Utils.py]: Is this really required?

**Comment by Tadly** [comment]
I added a few things I seem to have missed.
Squashing will be no issue :)

**Comment by enen92** [comment]
Just to make sure. Your script @BADMS is also in the jarvis branch. Do you intent to keep a separate version for Krypton?

**Comment by BADMS** [comment]
@enen92 it would break Jarvis skins that use it, as totally reworked daemon.

Hopefully all ok now, thanks all 👍

**Pull request merged**